### ISSUE TITLE
How to use compare with both signatures and *.sbt.json files? 

### ISSUE THREAD

**Issue by jolespin** [dib-lab/sourmash#672]
I'm working on a pipeline that uses `sourmash`.  I want to compare the signatures within a database plus [an] addition signature[s].  It doesn't look like I can do this currently.  I also can't assume that I will have the `signatures` available for the sequences in the database.

**Is there a flag I can use to do this?** 
or
**Is there a way to extract signatures from the database?** 

```
bash-4.1$ sourmash compare -k 31 -o out.txt bin.3.orig.fa.sig db.sbt.json
== This is sourmash version 2.0.0. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

Error in parsing signature; quitting.
Exception:

warning: no signatures loaded at given ksize/molecule type from db.sbt.json
loaded 1 signatures total.

0-output/M-1...	[ 1.]
min similarity in matrix: 1.000
saving labels to: out.txt.labels.txt
saving distance matrix to: out.txt
```

**Comment by ctb** [comment]
hi @jolespin we don't have a canonical way to get the signatures out of an SBT, but they _are_ all stored in individual files that you can retrieve if need be - e.g. for `podar-ref.sbt.json`, you will find all of the signatures under `.sbt.podar-ref/`, named as hashes. So you can do something like

`find .sbt.podar-ref \! -name internal.\* -print`

to get a list.

This is obviously suboptimal :)

Two ideas are --

* allow `compare` to take in SBT and LCA files.
* support a `sourmash signature` operation to extract signatures from SBT and LCA files.

**Comment by ctb** [comment]
#1044 provides `sourmash sig extract` which extracts signatures from SBT/LCA files; resolving #875 could provide a single loading API for compare.

**Comment by ctb** [comment]
This is fixed in #1059.
